src/services/users/userResetPassword.service.ts

```diff
@@ -0,0 +1,37 @@
+import dataSource from "../../data-source";
+import { Users } from "../../entities/users.entity";
+import { AppError } from "../../errors/appError";
+import jwt from "jsonwebtoken";
+import "dotenv/config"
+
+
+export const userResetPasswordService = async (token: string) => {
+     
+   
+   return jwt.verify(token,process.env.SECRET_KEY, async(error, decoded: any) => {
+      
+      if (error) {
+         throw new AppError(error.message, 409);
+      }
+
+      
+      const userRepository = dataSource.getRepository(Users)
+      const user = await userRepository.findOne({where: {id: decoded.sub, email: decoded.email}})
+     
+      if (!user.isActive) {
+         throw new AppError("User inactive", 409) 
```

Comment:
E para esses 2 acho melhor 401 dá para alterar aqui direto no review, se for o caso